src/main/java/org/mockito/internal/matchers/Matches.java

```diff
@@ -23,6 +26,8 @@ public Matches(Pattern pattern) {
 
     @Override
     public boolean matches(Object actual) {
+        Coverage.setTotalBranches("Matches::matches", 1);
+        Coverage.reached("Matches::matches", 0);
         return (actual instanceof String) && pattern.matcher((String) actual).find();
```

Comment:
It's a tricky one, I don't think you can do better, but technically the `&&` implies another brnch.

Reply:
It's posible to change the code in order to make a proper coverage analysis, but i think this it the best option (i'm not so happy with that)